hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/LegacyReplicationManager.java

```diff
@@ -2235,4 +2248,30 @@ private void replicateAnyWithTopology(ContainerInfo container,
           container.getContainerID(), ex);
     }
   }
+
+  private void closeEmptyContainer(ContainerInfo containerInfo) {
```

Comment:
Nit: if looking to continued support for the `legacyReplicationManager` it may be useful to add small unit test to `TestLegacyReplicationManager.java` to handle `CLOSING` lifestate for container with no replicas and no keys.